core/capabilities/compute/compute_test.go

```diff
@@ -132,7 +170,88 @@ func Test_Compute_Execute(t *testing.T) {
 			ReferenceID: "compute",
 		},
 	}
-	resp, err = compute.Execute(tests.Context(t), req)
+	resp, err = th.compute.Execute(tests.Context(t), req)
 	assert.NoError(t, err)
 	assert.False(t, resp.Value.Underlying["Value"].(*values.Bool).Underlying)
 }
+
+func TestComputeFetch(t *testing.T) {
+	workflowID := "15c631d295ef5e32deb99a10ee6804bc4af13855687559d7ff6552ac6dbb2ce0"
+	workflowExecutionID := "95ef5e32deb99a10ee6804bc4af13855687559d7ff6552ac6dbb2ce0abbadeed"
+	th := setup(t, defaultConfig)
+
+	th.connector.EXPECT().DonID().Return("don-id")
+	th.connector.EXPECT().GatewayIDs().Return([]string{"gateway1", "gateway2"})
+
+	msgID := strings.Join([]string{
+		workflowID,
+		workflowExecutionID,
+		ghcapabilities.MethodComputeAction,
+		validRequestUUID,
+	}, "/")
+
+	gatewayResp := gatewayResponse(t, msgID)
+	th.connector.On("SignAndSendToGateway", mock.Anything, "gateway1", mock.Anything).Return(nil).Run(func(args mock.Arguments) {
```

Comment:
`EXPECT()`

Reply:
here the `On` is used because we are interested in `Run` HandleGatewayMessage when this is called besides returning a response.
```
.Run(func(args mock.Arguments) {
		th.connectorHandler.HandleGatewayMessage(context.Background(), "gateway1", gatewayResp)
	}).Once()```
	
I think Expect is used mostly when checking for expectations and returning a value.